pkg/services/kubeconfig_loader.go

```diff
@@ -0,0 +1,239 @@
+		k.logger.Debugf("No changes detected in kubeconfig %s, skipping reload", k.path)
+		return
+	}
+	k.logger.Debugf("Detected kubeconfig change, reloading: %s", k.path)
+	for client := range k.remoteKubeClients { // if we reassign the map, the synchronizers lose the reference to it
+		delete(k.remoteKubeClients, client)
+		k.logger.Debugf("Removed remote client for cluster: %s", client)
+	}
+
+	// Update stored hash
+	k.lastConfigHash = currentHash
+
+	kubeConfig, err := clientcmd.LoadFromFile(k.path)
+	if err != nil { // kubeConfig is nil only if the file is empty or invalid
+		k.logger.Errorf("Error loading kube config from path %s: %s", k.path, err)
+		return
+	}
+	if len(kubeConfig.Contexts) == 0 {
+		k.logger.Warnf("No contexts found in kubeconfig file: %s", k.path)
+		return
+	}
+
+	var remoteClustersName []string
+	for contextName := range kubeConfig.Contexts {
+		remoteClustersName = append(remoteClustersName, contextName)
+	}
+	k.logger.Debugf("Remote clusters found in kubeconfig: %v", remoteClustersName)
+
+	if len(remoteClustersName) > 0 {
+		var initializedClustersName []string
+		for _, cluster := range remoteClustersName {
+			remoteClusterConfig, err := BuildConfigWithContextFromFlags(cluster, k.path)
+			if err != nil {
+				k.logger.Errorf("Error building kubeconfig for cluster '%s': %s", cluster, err)
+				continue
+			}
+
+			remoteClusterConfig.Timeout = 1 * time.Second // Set a timeout for the HTTP client, maybe this should be configurable
+			var remoteClusterClient IKubeClient
+			if k.clientFactory != nil {
+				remoteClusterClient, err = k.clientFactory(remoteClusterConfig)
+			} else {
+				remoteClusterClient, err = kubernetes.NewForConfig(remoteClusterConfig)
+				remoteClusterClient = newKubeClientAdapter(remoteClusterClient.(*kubernetes.Clientset))
+			}
+			if err != nil {
+				k.logger.Errorf("Error creating remote clientset for cluster '%s': %s", cluster, err)
+				continue
+			}
+			output, err := remoteClusterClient.ServerVersion()
+			// This is a simple way to check if the server is reachable and the config is valid
+			if err != nil {
+				k.logger.Errorf("Error getting server version for cluster '%s': %s", cluster, err)
+				continue
+			}
+			k.logger.Infof("Connected to remote cluster '%s' with server version: %s", cluster, output.String())
+
+			k.remoteKubeClients[cluster] = remoteClusterClient
+			initializedClustersName = append(initializedClustersName, cluster)
+			k.logger.Debugf("Initialized remote cluster client for '%s'", cluster)
+		}
+
+		if len(initializedClustersName) > 0 {
+			k.logger.Infof("Remote clusters successfully initialized: %v", initializedClustersName)
+		}
+	}
+}
+
+// StartWatching starts watching the kubeconfig file for changes, including deletions and recreations.
+func (k *KubeconfigLoader) StartWatching(ctx context.Context) {
+	var debounceTimer *time.Timer // timer to debounce events and avoid multiple reloads
+	debounceDuration := time.Duration(500) * time.Millisecond
```

Comment:
Also here, same logic, sometimes is hard to find hardcoded variables to fine tune. Default value could be 500ms, but could be optional and configurable.

Reply:
Fixed! Added a configurable flag parameter.